qt/aqt/mediasrv.py

```diff
@@ -752,23 +746,11 @@ def legacy_page_data() -> Response:
         return _text_response(HTTPStatus.NOT_FOUND, "page not found")
 
 
-def _extract_page_context() -> PageContext:
-    "Get context based on referer header."
-    from urllib.parse import parse_qs, urlparse
-
-    referer = urlparse(request.headers.get("Referer", ""))
-    if referer.path.startswith("/_anki/pages/") or is_sveltekit_page(referer.path[1:]):
-        return PageContext.NON_LEGACY_PAGE
-    elif referer.path == "/_anki/legacyPageData":
-        query_params = parse_qs(referer.query)
-        query_id = query_params.get("id")
-        if not query_id:
-            return PageContext.UNKNOWN
-        id = int(query_id[0])
-        page_context = aqt.mw.mediaServer.get_page_context(id)
-        return page_context if page_context else PageContext.UNKNOWN
-    else:
-        return PageContext.UNKNOWN
+_APIKEY = "".join(random.choices(string.ascii_letters + string.digits, k=32))
```

Comment:
Sorry for intruding, but it better be [`secrets.token_urlsafe`](https://docs.python.org/3/library/secrets.html#secrets.token_urlsafe). `random` is more or less predictable:
- https://blog.securityevaluators.com/hacking-the-javascript-lottery-80cc437e3b7f
- https://blog.securityevaluators.com/xorshift128-backward-ff3365dc0c17
- https://lemire.me/blog/2017/08/22/cracking-random-number-generators-xoroshiro128/
- https://github.com/deut-erium/RNGeesus

Reply:
Sorry, I lost track of this. This change will be in the next patch release.